syntax = "proto3";

package statistic;

service StatisticService {
    rpc GetStatisticByPost(GetByPostRequest) returns (GetByPostResponse);
    rpc GetPopularPost(GetPopularPostRequest) returns (GetPopularPostResponse);
    rpc GetPopularUser(NoParams) returns (GetPopularUserResponse);
}

message NoParams {
    
}

message GetByPostRequest {
    string id = 1;
}

message GetByPostResponse {
    int64 likes = 1;
    int64 views = 2; 
}

message GetPopularPostRequest {
    string sort_type = 1;
}

message PostAnswer {
    string post_id = 1;
    string author = 2;
    int64 count = 3;
}

message GetPopularPostResponse {
    repeated PostAnswer popular_posts = 1;
}

message UserAnswer {
    string author = 1;
    int64 likes = 2;
}

message GetPopularUserResponse {
    repeated UserAnswer popular_users = 1;
}